packages/go_router/tool/run_tests.dart

```diff
@@ -9,27 +9,86 @@
 // ignore_for_file: avoid_print
 
 import 'dart:io';
+import 'package:io/io.dart' as io;
 import 'package:path/path.dart' as p;
+import 'package:yaml_edit/yaml_edit.dart';
 
 Future<void> main(List<String> args) async {
-  if (!Platform.isMacOS) {
-    print('This test can only be run on macOS.');
+  if (!Platform.isMacOS /* && !Platform.isWindows*/) {
+    print('This test can only be run on macOS' /*' or windows.'*/);
     exit(0);
   }
+  final p.Context ctx = p.context;
   final Directory packageRoot =
-      Directory(p.dirname(Platform.script.path)).parent;
+      Directory(ctx.dirname(_ensureTrimLeadingSeparator(Platform.script.path)))
+          .parent;
+
+  //copy from go_router/test_fixes to temp directory
+  final Directory testFixesTargetDir = await Directory.systemTemp.createTemp();
+
+  await _prepareTemplate(
+    ctx: ctx,
+    testFixesDir: ctx.join(packageRoot.path, 'test_fixes'),
+    testFixesTargetDir: testFixesTargetDir.path,
+    //Switching this to true, will depend on go_router v7.0.0, which will fail the golden test
+    testBadVersion: false,
```

Comment:
We should always write test that test the latest version. That being said, I am not sure if this change can be testable as presubmit as the change would need to wait for the change to be merged and published in a new version if we were to test the dart fix with go_router from pub dev.

Reply:
I think we should revert this change and ask for no test exception

Reply:
the modified test, will simulate a pub.dev install, that is, a dependency from outside the original pubspec.yaml scope